src/routing_table/prefix.rs

```diff
@@ -73,6 +73,12 @@ impl<T: Clone + Copy + Default + Binary + Xorable> Prefix<T> {
         i >= self.bit_count() || i >= other.bit_count()
     }
 
+    /// Returns `true` if `other` is compatible but strictly shorter than `self`.
+    pub fn extends(&self, other: &Prefix<T>) -> bool {
```

Comment:
I think it's weird too, I would have thought `is_prefix_of` with the arguments switched would be the most natural.